openmc/filter.py

```diff
@@ -162,12 +187,12 @@ def bins(self, bins):
                     raise ValueError(msg)
 
         # mesh filters
-        elif self._type == 'mesh':
+        elif self.type == 'mesh':
             if not len(bins) == 1:
                 msg = 'Unable to add bins "{0}" to a mesh Filter since ' \
                       'only a single mesh can be used per tally'.format(bins)
                 raise ValueError(msg)
-            elif not _isinstance(bins[0], Integral):
+            elif not cv._isinstance(bins[0], Integral):
```

Comment:
I agree with your comment above, but in this case the bin may be a `numpy.int` which would error out for NumPy <v1.9 with `cv.instance(...)`.